sno/merge.py

```diff
@@ -35,11 +43,31 @@
     help="Don't perform a merge - just show what would be done",
 )
 @click.argument("commit", required=True, metavar="COMMIT")
+@do_json_option
 @click.pass_context
-def merge(ctx, ff, ff_only, dry_run, commit):
+def merge(ctx, ff, ff_only, dry_run, do_json, commit):
     """ Incorporates changes from the named commits (usually other branch heads) into the current branch. """
     repo = ctx.obj.repo
+    merge_jdict = do_merge(repo, ff, ff_only, dry_run, commit)
+    no_op = merge_jdict.get("no-op", False) or merge_jdict.get("dryRun", False)
+
+    if not no_op:
+        # Update working copy.
+        repo_structure = RepositoryStructure(repo)
+        wc = repo_structure.working_copy
+        L.debug(f"Updating {wc.path} ...")
+        merge_commit = repo[merge_jdict["mergeCommit"]]
+        wc.reset(merge_commit, repo_structure)
```

Comment:
Can you merge without a working copy? Should be possible, right? If so this needs to either handle it or raise NotYetImplemented or something

Reply:
For now, just fixed this so it won't crash if you have no working copy.

Potentially needs more work later: a user who is halfway through a merge with lots of conflicts to resolve should probably not be messing with their working copy, since this is a good way to a) get very confused and b) lose updates. (Just like it isn't a good idea to be editing files in your editor at the same time as you update them on disk using a different program).

So, we may want to hide or lock the working copy for the duration of a merge. But this is not done in this PR

Reply:
ok 👍 

It's a good way to get confused yes. I think _removing_ the WC would be even more confusing. Locking it would be reasonable, might work for GPKG (not sure) but isn't going to work for future WC types (postgres)

WWGD? I think git would probably do the commit but then refuse to update the working copy afterwards. Then you'd need to type some special git reset command to allow it to reset to the new commit.

Reply:
ah - it's different for git since the WC is meaningful during a merge, since it's used for resolving conflicts. Even though you have to add the resovled conflicts to the staging area, the WC is otherwise at the same state as the resolved merge. That isn't the case in sno

Reply:
1. removing the WC isn't an option.
2. but we're not pushing conflicts through the WC though, right? So can't we just ignore it and basically refuse to look at it until the merge is completed or aborted? (similarly I've seen git give some "you're in the middle of merging, I can't do that!" errors)